tests/test_react_auth_defs.py

```diff
@@ -52,11 +55,45 @@
 DOMAIN = "some-domain"
 # Mock out the call which create_authtoken makes Envs.get_user_auth_info
 # to get the allowed environments and first/last name for the user.
-ENVS.get_user_auth_info = lambda email: (ALLOWED_ENVS, FIRST_NAME, LAST_NAME)
+ENVS.get_user_auth_info = lambda email, raise_exception: (ALLOWED_ENVS, FIRST_NAME, LAST_NAME)
 
 AUTH = Auth(AUTH0_CLIENT_ID, AUTH0_SECRET, ENVS)
 
 
+class MockChaliceApp:
+
+    class MockReactApi:
+        def __init__(self):
+            self.react_authorize = AUTH.authorize
+            self.create_success_response = ReactApi.create_success_response
+            self.create_response = ReactApi.create_response
+            self.get_site_name = lambda: "some-site-name"
+            self.APP_PACKAGE_NAME = "foursight"
+
+    class MockChaliceRequest:
+        def __init__(self, request: dict) -> None:
+            self._request = request
+
+        def to_dict(self) -> dict:
+            return self._request
+
+    def __init__(self, current_request: Optional[dict] = None):
+        self._current_request = MockChaliceApp.MockChaliceRequest(current_request)
+
+    @property
+    def current_request(self) -> MockChaliceRequest:
+        return self._current_request
+
+    @property
+    def core(self) -> MockReactApi:
+        return MockChaliceApp.MockReactApi()
+
+    def route(self, path: str, **kwargs):
+        def route_registration(wrapped_route_function):
+            return wrapped_route_function
+        return route_registration
+
+
 def create_test_jwt_unencoded():
```

Comment:
Should some or all of these `create` functions be methods of `MockChaliceApp`. It might not matter. I was just pondering how these are grouped conceptually. It's fine to leave them.